Themis/Views/Assessment/FiletreeSidebar/FiletreeSidebarView.swift

```diff
@@ -46,4 +50,39 @@ struct FiletreeSidebarView: View {
         }
         .frame(maxWidth: .infinity, alignment: .leading)
     }
+    @ViewBuilder
+    func fileView(tree: Node) -> some View {
+        HStack {
+            iconView(tree: tree)
+            Text(tree.name)
+        }
+    }
+    
+    @ViewBuilder
+    func iconView(tree: Node) -> some View {
+        Group {
+            if let language = tree.language(), let image = image(for: language, style: .original) {
+                image
+                    .resizable()
+                    .frame(width: 20, height: 20)
+            } else {
+                Spacer()
+                    .frame(width: 30)
+            }
+        }
+    }
+    
+    func image(for language: Language, style: Style) -> Image? {
+        let imageName = "\(language.rawValue)-\(style.rawValue)"
+        if let image = UIImage(named: imageName){
+            return Image(uiImage: image)
+        }
+        let alternativeStyle = style == .plain ? Style.original : .plain
+        let alternativeImageName = "\(language.rawValue)-\(alternativeStyle.rawValue)"
+        if let image = UIImage(named: alternativeImageName) {
+            return Image(uiImage: image)
+        }
+        return nil
+    }
+
```

Comment:
I feel like there's enough logic in here that would justify moving this into its own component => own file.

Reply:
Ok I extracted it